src/pvsite_forecast.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime, timedelta, time, timezone
+import streamlit as st
+import os
+from pvsite_datamodel.connection import DatabaseConnection
+from pvsite_datamodel.sqlmodels import SiteSQL, ForecastSQL, ForecastValueSQL, GenerationSQL
+from pvsite_datamodel.read import (
+    get_all_sites,
+    get_pv_generation_by_sites,
+    get_latest_forecast_values_by_site
+)
+
+import plotly.graph_objects as go
+
+
+colour_per_model = {
+    # "cnn": "#FFD053",
+    # "National_xg": "#7BCDF3",
+    # "pvnet_v2": "#4c9a8e",
+    # "PVLive Initial Estimate": "#e4e4e4",
+    # "PVLive Updated Estimate": "#e4e4e4",
+    # "PVLive GSP Sum Estimate": "#FF9736",
+    # "PVLive GSP Sum Updated": "#FF9736",
+}
+
+
+def pvsite_forecast_page():
+    """Main page for pvsite forecast"""
+    st.markdown(
+        f'<h1 style="color:#FFD053;font-size:48px;">{"OCF Analysis Dashboard"}</h1>',
+        unsafe_allow_html=True,
+    )
+    st.markdown(
+        f'<h1 style="color:#63BCAF;font-size:48px;">{"PV Site Forecast"}</h1>',
+        unsafe_allow_html=True,
+    )
+    # get site_uuids from database
+
+    url = os.env("SITES_DB_URL")
+    connection = DatabaseConnection(url=url, echo=True)
+    with connection.get_session() as session:
+        site_uuids = get_all_sites(session=session)
+        site_uuids = [
+            sites.site_uuid for sites in site_uuids if sites.site_uuid is not None
+        ]
+      
+    site_selection = st.sidebar.selectbox("Select sites by site_uuid", site_uuids, index=0)
+    starttime = st.sidebar.date_input("Start Date", datetime.today() - timedelta(days=3))
+    st.write("Forecast for", site_selection)
+    # get forecast values for selected sites and plot
+    with connection.get_session() as session:
+        forecasts = get_latest_forecast_values_by_site(
+            session=session,
+            site_uuids=[site_selection],
+            start_utc=starttime,
+        )
+
+        forecasts = forecasts.values()
+        for forecast in forecasts:
+            for i in forecast: 
+                print(i.forecast_power_kw, i.created_utc)
+
+    with connection.get_session() as session:
+        generations = get_pv_generation_by_sites(
+            session=session,
+            site_uuids=[site_selection],
+            start_utc=starttime,
+        )
+        # generations = [int(generation.generation_power_kw) for generation in generations if generation is not None]
+        # print("generations", generations)
+
+        yy = [generation.generation_power_kw for generation in generations if generation is not None]
+        xx = [generation.created_utc for generation in generations]
```

Comment:
I think it might be better to use `datetime_utc` or something like that,r ather than the created_utc

Reply:
would `start_utc` work? that seems to be what we use in the pvsite api when we convert the value to pydantic: https://github.com/openclimatefix/pv-site-api/blob/main/pv_site_api/_db_helpers.py#L96
<img width="687" alt="Screenshot 2023-08-24 at 10 08 42" src="https://github.com/openclimatefix/uk-analysis-dashboard/assets/86949265/d288cb70-e216-4fd1-b476-5ed362ed537e">

Reply:
I realise the above code is for generation, but it's the same for generation in the pvsite api, using `start_utc` as the generation timestamp:
<img width="585" alt="Screenshot 2023-08-24 at 10 13 20" src="https://github.com/openclimatefix/uk-analysis-dashboard/assets/86949265/528b4b09-ce06-4b78-8503-7bad72678306">

Reply:
The date change seems to be working. The chart goes slightly into the future now, and I think yesterday nothing was in the future.
<img width="1023" alt="Screenshot 2023-08-24 at 10 29 29" src="https://github.com/openclimatefix/uk-analysis-dashboard/assets/86949265/52976930-5b35-4bf5-a317-23debf31a171">

Reply:
Yea, that looks good